wire-library/wire-java-generator/src/test/java/com/squareup/wire/java/JavaGeneratorTest.java

```diff
@@ -609,6 +609,27 @@ public void generateTypeUsesPackageNameOnFieldAndClassNameClashWithinPackage() t
     assertThat(code).contains("class Person");
   }
 
+  @Test public void packageNameUsedIfFieldNameIsSameAsNonScalarTypeName() throws IOException {
+    RepoBuilder repoBuilder = new RepoBuilder()
+            .add("common/common_message.proto",
+                    "package a.Common;\n"
+                            + "option java_package = \"a.common\";"
+                            + "message CommonMessage {\n"
+                            + "	required string First = 1;\n"
+                            + "}\n")
+            .add("example.proto",
+                    "package a;\n"
+                            + "import \"common/common_message.proto\";\n"
+                            + "\n"
+                            + "message Example {\n"
+                            + "	required Common.CommonMessage CommonMessage = 1;\n"
+                            + "}\n");
```

Comment:
I locally formatted it for better reading.
```suggestion
        .add("common/common_message.proto",
            "package a.Common;\n"
          + "option java_package = \"a.common\";"
          + "message CommonMessage {\n"
          + "   required string First = 1;\n"
          + "}\n")
        .add("example.proto",
            "package a;\n"
          + "import \"common/common_message.proto\";\n"
          + "\n"
          + "message Example {\n"
          + "   required Common.CommonMessage CommonMessage = 1;\n"
          + "}\n");
```